compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -2133,8 +2133,45 @@ trait Applications extends Compatibility {
     }
   }
 
-  def isApplicableExtensionMethod(ref: TermRef, receiver: Type)(using Context) =
-    ref.symbol.is(ExtensionMethod)
-    && !receiver.isBottomType
-    && isApplicableMethodRef(ref, receiver :: Nil, WildcardType)
+  /** Assuming methodRef is a reference to an extension method defined e.g. as
+   *
+   *  extension [T1, T2](using A)(using B, C)(receiver: R)(using D)
+   *    def foo[T3](using E)(f: F): G = ???
+   *
+   *  return the tree representing methodRef partially applied to the receiver and all the implicit parameters preceding it (A, B, C)
+   *  with the type parameters of the extension (T1, T2) inferred.
+   *  A failure is returned if the implicit search fails for any of the leading implicit parameters or if the receiver has a wrong type
+   *  (note that in general the type of the receiver might depend on the exact types of the found instances of the proceding implicits).
+   *  No implicit search is tried for implicits following the receiver or for parameters of the def (D, E).
+   */
+  private def applyWithoutPostreceiverImplicits(methodRef: TermRef, receiver: Tree)(using Context): scala.util.Try[Tree] =
+      // Drop all parameters sections of an extension method following the receiver; the return type after truncation is not important
+    def truncateExtension(tp: Type)(using Context): Type = tp match
+      case poly: PolyType => poly.newLikeThis(poly.paramNames, poly.paramInfos, truncateExtension(poly.resType))
+      case meth: MethodType if meth.isContextualMethod => meth.newLikeThis(meth.paramNames, meth.paramInfos, truncateExtension(meth.resType))
+      case meth: MethodType => meth.newLikeThis(meth.paramNames, meth.paramInfos, defn.AnyType)
+
+    val truncatedSym = methodRef.symbol.asTerm.copy(owner = defn.RootPackage, name = Names.termName(""), info = truncateExtension(methodRef.info))
```

Comment:
It seems to me `.asTerm` is not useful:

```suggestion
    val truncatedSym = methodRef.symbol.copy(owner = defn.RootPackage, name = Names.termName(""), info = truncateExtension(methodRef.info))
```

Reply:
Actually it is seems to be necessary as for some reason `methodRef.symbol` is just a `Symbol`, not a `TermSymbol`